images.CI/linux-and-win/convert-to-vhd.ps1

```diff
@@ -0,0 +1,211 @@
+  --resource-group $ResourceGroupName `
+  --name $ManagedImageName `
+  --duration-in-seconds 86400 `
+  --access-level Read `
+  --query [accessSas] -o tsv
+
+Write-Host "Generating SAS URL for the destination storage account '$StorageAccountName'..."
+$destinationVHDBlobUri = Get-DestinationBlobUri `
+  -ContainerName $StorageAccountContainerName `
+  -SubscriptionId $SubscriptionId `
+  -ResourceGroupName $ResourceGroupName `
+  -StorageAccountName $StorageAccountName `
+  -TenantId $TenantId `
+  -ImageBlobName $VhdName
+
+Write-Host "Copying VHD blob from '$($sourceDiskUri.Split('?')[0])' to '$($destinationVHDBlobUri.Split('?')[0])'..."
+
+$jobLog = New-Object 'System.Collections.Generic.List[PSCustomObject]'
+Write-Host 'Starting azcopy'
+
+& $env:AZCOPYPATH copy $sourceDiskUri $DestinationVHDBlobUri --s2s-preserve-access-tier=false --output-type json | Foreach-Object {
+  # Parse Json
+  $json = ConvertFrom-Json -InputObject $_
+  if ($json.MessageContent.StartsWith('{')) {
+    $json.MessageContent = ConvertFrom-Json -InputObject $json.MessageContent
+  }
+  $jobLog.Add($json)
+
+  # Output current status
+  if ($json.MessageType -in @('Progress', 'EndOfJob')) {
+    $mc = $json.MessageContent
+    $status = $mc.JobStatus
+    $bytesCopied = [int]($mc.TotalBytesTransferred / 1MB)
+    $totalBytes = [int]($mc.TotalBytesExpected / 1MB)
+    $percentComplete = [Math]::Round($mc.PercentComplete, 2)
+    # Write-Host "$env:imageBlobName -> $targetAccountName : $status [$percentComplete %]: $bytesCopied/$totalBytes MB Copied..."
+    Write-Host " $status [$percentComplete %]: $bytesCopied/$totalBytes MB Copied..."
+  }
+}
+
+Write-Host 'azcopy finished'
+
+# Check last exit code
+if ($LastExitCode) {
+  Write-Host $LastExitCode
+
+  $failedJob = $jobLog | Where-Object { $_.MessageType -eq 'EndOfJob' -and $_.MessageContent.JobStatus -eq 'Failed' }
+
+  if ($failedJob) {
+    $errorContent = $failedJob.MessageContent.FailedTransfers.ErrorCode
+  } else {
+    $errorContent = ($jobLog | Where-Object MessageType -eq 'Error').MessageContent
+  }
+  #throw "Copy [$env:imageBlobName] failed with error: $errorContent"
+  throw "Blob Copy failed with error: $errorContent"
+}
+
+Write-Host "Successfully converted '$ManagedImageName' to '$VhdName' in '$StorageAccountName' storage account."
+Write-Host "Cleaning up..."
+
+# Revoke SAS URL for the Managed Disk
+az disk revoke-access `
+  --resource-group $ResourceGroupName `
+  --name $ManagedImageName `
+  --only-show-errors
+if ($LastExitCode) {
+  Write-Host "Warning: Failed to revoke access to the Managed Disk '$ManagedImageName'."
+}
+
+# Delete Azure Managed Disk from Shared Image Gallery
+az disk delete `
+  --resource-group $ResourceGroupName `
+  --name $ManagedImageName `
+  --only-show-errors `
+  --yes
+if ($LastExitCode) {
+  Write-Host "Warning: Failed to delete the Managed Disk '$ManagedImageName'."
+}
+
+# Delete Image Version from Shared Image Gallery
```

Comment:
I think we can leave the version, do we have gallery limits? cc @Alexey-Ayupov

Reply:
What do you mean? If we just print value here, then nobody will track the value and hitting the limit won't be expected by anyone.

Reply:
Ok. Maybe we should set the retention period to 7 days and delete image versions after the date expiration? For example `Get-AzGalleryImageVersion` returns "PublishedDate" that can be used.

Reply:
I think it's better to implement this outside of build pipeline. Don't we already have something to remove outdated resources from our subscription?

Reply:
Anyway, why do we even want to keep image version in gallery?